File: app/src/main/java/com/lumiyaviewer/lumiya/base64/Base64.java
/*
 * Clean Base64 implementation for Linkpoint
 * Replaces problematic decompiled code with working implementation
 */
package com.lumiyaviewer.lumiya.base64;

/**
 * Base64 encoding/decoding utility using Android's built-in Base64 implementation
 */
public class Base64 {
    
    /**
     * Decode a Base64 encoded string to bytes
     */
    public static byte[] decode(String input) {
        if (input == null) {
            return new byte[0];
        }
        return android.util.Base64.decode(input, android.util.Base64.DEFAULT);
    }
    
    /**
     * Encode bytes to a Base64 string
     */
    public static String encode(byte[] input) {
        if (input == null) {
            return "";
        }
        return android.util.Base64.encodeToString(input, android.util.Base64.DEFAULT);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/base64/Base64.kt
package com.lumiyaviewer.lumiya.base64

import android.util.Base64 as AndroidBase64

/**
 * Modern Kotlin Base64 utility
 * Provides encoding/decoding functionality using Android's built-in Base64
 */
object Base64 {
    
    /**
     * Decodes a Base64 encoded string to bytes
     * @param input The Base64 encoded string
     * @return Decoded byte array, or empty array if input is null
     */
    @JvmStatic
    fun decode(input: String?): ByteArray {
        return input?.let { 
            AndroidBase64.decode(it, AndroidBase64.DEFAULT) 
        } ?: byteArrayOf()
    }
    
    /**
     * Encodes bytes to a Base64 string
     * @param input The byte array to encode
     * @return Base64 encoded string, or empty string if input is null
     */
    @JvmStatic
    fun encode(input: ByteArray?): String {
        return input?.let { 
            AndroidBase64.encodeToString(it, AndroidBase64.DEFAULT) 
        } ?: ""
    }
    
    /**
     * Decodes a Base64 encoded string to bytes with specified flags
     * @param input The Base64 encoded string
     * @param flags Android Base64 flags
     * @return Decoded byte array
     */
    @JvmStatic
    fun decode(input: String?, flags: Int): ByteArray {
        return input?.let { 
            AndroidBase64.decode(it, flags) 
        } ?: byteArrayOf()
    }
    
    /**
     * Encodes bytes to a Base64 string with specified flags
     * @param input The byte array to encode
     * @param flags Android Base64 flags
     * @return Base64 encoded string
     */
    @JvmStatic
    fun encode(input: ByteArray?, flags: Int): String {
        return input?.let { 
            AndroidBase64.encodeToString(it, flags) 
        } ?: ""
    }
}